File: Problems/1762. Buildings With an Ocean View.cpp
class Solution {
public:
    vector<int> findBuildings(vector<int>& heights) {
        int maxHeight = -1;
        vector<int> oceanViewBuildings;
        
        for(int i=heights.size()-1; i>=0; i--){
            if(heights[i] > maxHeight){
                oceanViewBuildings.push_back(i);
            }
            maxHeight = max(maxHeight, heights[i]);
        }
        
        reverse(oceanViewBuildings.begin(), oceanViewBuildings.end());
        
        return oceanViewBuildings;
    }
};

File: Problems/Python/1762. Buildings With an Ocean View.py
class Solution(object):
    def findBuildings(self, heights):
        """
        :type heights: List[int]
        :rtype: List[int]
        """
        oceanView = []
        maxHeight = -1
        counter = len(heights) - 1
        
        for i in reversed(heights):
            if i > maxHeight:
                oceanView.insert(0, counter)
            maxHeight = max(maxHeight, i)
            counter = counter - 1
        
        return oceanView
